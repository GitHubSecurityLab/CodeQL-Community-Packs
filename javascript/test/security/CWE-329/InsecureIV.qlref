security/CWE-329/InsecureIV.ql
