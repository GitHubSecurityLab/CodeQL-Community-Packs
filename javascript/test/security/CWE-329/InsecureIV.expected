nodes
| examples/secure_iv.js:11:7:11:14 | randomIV |
| examples/secure_iv.js:11:7:11:44 | randomIV |
| examples/secure_iv.js:11:7:11:44 | randomIV |
| examples/secure_iv.js:11:18:11:44 | crypto. ... eysize) |
| examples/secure_iv.js:11:18:11:44 | crypto. ... eysize) |
| examples/secure_iv.js:13:63:13:62 | randomIV |
| examples/secure_iv.js:14:54:14:61 | randomIV |
| examples/secure_iv.js:14:54:14:61 | randomIV |
| examples/secure_iv_tainted.js:11:7:11:14 | randomIV |
| examples/secure_iv_tainted.js:11:7:11:76 | randomIV |
| examples/secure_iv_tainted.js:11:7:11:76 | randomIV |
| examples/secure_iv_tainted.js:11:7:11:76 | randomIV |
| examples/secure_iv_tainted.js:11:18:11:39 | crypto. ... tes(32) |
| examples/secure_iv_tainted.js:11:18:11:39 | crypto. ... tes(32) |
| examples/secure_iv_tainted.js:11:18:11:48 | crypto. ... oString |
| examples/secure_iv_tainted.js:11:18:11:48 | crypto. ... oString |
| examples/secure_iv_tainted.js:11:18:11:58 | crypto. ... ase64') |
| examples/secure_iv_tainted.js:11:18:11:58 | crypto. ... ase64') |
| examples/secure_iv_tainted.js:11:18:11:58 | crypto. ... ase64') |
| examples/secure_iv_tainted.js:11:18:11:58 | crypto. ... ase64') |
| examples/secure_iv_tainted.js:11:18:11:64 | crypto. ... ).slice |
| examples/secure_iv_tainted.js:11:18:11:64 | crypto. ... ).slice |
| examples/secure_iv_tainted.js:11:18:11:76 | crypto. ... eysize) |
| examples/secure_iv_tainted.js:11:18:11:76 | crypto. ... eysize) |
| examples/secure_iv_tainted.js:11:18:11:76 | crypto. ... eysize) |
| examples/secure_iv_tainted.js:11:18:11:76 | crypto. ... eysize) |
| examples/secure_iv_tainted.js:13:63:13:62 | randomIV |
| examples/secure_iv_tainted.js:14:54:14:61 | randomIV |
| examples/secure_iv_tainted.js:14:54:14:61 | randomIV |
| examples/secure_iv_tainted.js:14:54:14:61 | randomIV |
| examples/secure_iv_tainted.js:14:54:14:61 | randomIV |
| examples/static_iv.js:11:7:11:34 | fixedIV |
| examples/static_iv.js:11:17:11:34 | "0123456789abcdef" |
| examples/static_iv.js:11:17:11:34 | "0123456789abcdef" |
| examples/static_iv.js:14:54:14:60 | fixedIV |
| examples/static_iv.js:14:54:14:60 | fixedIV |
edges
| examples/secure_iv.js:11:7:11:44 | randomIV | examples/secure_iv.js:11:7:11:14 | randomIV |
| examples/secure_iv.js:11:7:11:44 | randomIV | examples/secure_iv.js:13:63:13:62 | randomIV |
| examples/secure_iv.js:11:7:11:44 | randomIV | examples/secure_iv.js:14:54:14:61 | randomIV |
| examples/secure_iv.js:11:7:11:44 | randomIV | examples/secure_iv.js:14:54:14:61 | randomIV |
| examples/secure_iv.js:11:18:11:44 | crypto. ... eysize) | examples/secure_iv.js:11:7:11:44 | randomIV |
| examples/secure_iv.js:11:18:11:44 | crypto. ... eysize) | examples/secure_iv.js:11:7:11:44 | randomIV |
| examples/secure_iv.js:11:18:11:44 | crypto. ... eysize) | examples/secure_iv.js:11:7:11:44 | randomIV |
| examples/secure_iv.js:11:18:11:44 | crypto. ... eysize) | examples/secure_iv.js:11:7:11:44 | randomIV |
| examples/secure_iv_tainted.js:11:7:11:76 | randomIV | examples/secure_iv_tainted.js:11:7:11:14 | randomIV |
| examples/secure_iv_tainted.js:11:7:11:76 | randomIV | examples/secure_iv_tainted.js:13:63:13:62 | randomIV |
| examples/secure_iv_tainted.js:11:7:11:76 | randomIV | examples/secure_iv_tainted.js:14:54:14:61 | randomIV |
| examples/secure_iv_tainted.js:11:7:11:76 | randomIV | examples/secure_iv_tainted.js:14:54:14:61 | randomIV |
| examples/secure_iv_tainted.js:11:7:11:76 | randomIV | examples/secure_iv_tainted.js:14:54:14:61 | randomIV |
| examples/secure_iv_tainted.js:11:7:11:76 | randomIV | examples/secure_iv_tainted.js:14:54:14:61 | randomIV |
| examples/secure_iv_tainted.js:11:18:11:39 | crypto. ... tes(32) | examples/secure_iv_tainted.js:11:18:11:48 | crypto. ... oString |
| examples/secure_iv_tainted.js:11:18:11:39 | crypto. ... tes(32) | examples/secure_iv_tainted.js:11:18:11:48 | crypto. ... oString |
| examples/secure_iv_tainted.js:11:18:11:39 | crypto. ... tes(32) | examples/secure_iv_tainted.js:11:18:11:48 | crypto. ... oString |
| examples/secure_iv_tainted.js:11:18:11:39 | crypto. ... tes(32) | examples/secure_iv_tainted.js:11:18:11:48 | crypto. ... oString |
| examples/secure_iv_tainted.js:11:18:11:39 | crypto. ... tes(32) | examples/secure_iv_tainted.js:11:18:11:58 | crypto. ... ase64') |
| examples/secure_iv_tainted.js:11:18:11:39 | crypto. ... tes(32) | examples/secure_iv_tainted.js:11:18:11:58 | crypto. ... ase64') |
| examples/secure_iv_tainted.js:11:18:11:39 | crypto. ... tes(32) | examples/secure_iv_tainted.js:11:18:11:58 | crypto. ... ase64') |
| examples/secure_iv_tainted.js:11:18:11:39 | crypto. ... tes(32) | examples/secure_iv_tainted.js:11:18:11:58 | crypto. ... ase64') |
| examples/secure_iv_tainted.js:11:18:11:58 | crypto. ... ase64') | examples/secure_iv_tainted.js:11:18:11:64 | crypto. ... ).slice |
| examples/secure_iv_tainted.js:11:18:11:58 | crypto. ... ase64') | examples/secure_iv_tainted.js:11:18:11:64 | crypto. ... ).slice |
| examples/secure_iv_tainted.js:11:18:11:58 | crypto. ... ase64') | examples/secure_iv_tainted.js:11:18:11:76 | crypto. ... eysize) |
| examples/secure_iv_tainted.js:11:18:11:58 | crypto. ... ase64') | examples/secure_iv_tainted.js:11:18:11:76 | crypto. ... eysize) |
| examples/secure_iv_tainted.js:11:18:11:58 | crypto. ... ase64') | examples/secure_iv_tainted.js:11:18:11:76 | crypto. ... eysize) |
| examples/secure_iv_tainted.js:11:18:11:58 | crypto. ... ase64') | examples/secure_iv_tainted.js:11:18:11:76 | crypto. ... eysize) |
| examples/secure_iv_tainted.js:11:18:11:76 | crypto. ... eysize) | examples/secure_iv_tainted.js:11:7:11:76 | randomIV |
| examples/secure_iv_tainted.js:11:18:11:76 | crypto. ... eysize) | examples/secure_iv_tainted.js:11:7:11:76 | randomIV |
| examples/secure_iv_tainted.js:11:18:11:76 | crypto. ... eysize) | examples/secure_iv_tainted.js:11:7:11:76 | randomIV |
| examples/secure_iv_tainted.js:11:18:11:76 | crypto. ... eysize) | examples/secure_iv_tainted.js:11:7:11:76 | randomIV |
| examples/static_iv.js:11:7:11:34 | fixedIV | examples/static_iv.js:14:54:14:60 | fixedIV |
| examples/static_iv.js:11:7:11:34 | fixedIV | examples/static_iv.js:14:54:14:60 | fixedIV |
| examples/static_iv.js:11:17:11:34 | "0123456789abcdef" | examples/static_iv.js:11:7:11:34 | fixedIV |
| examples/static_iv.js:11:17:11:34 | "0123456789abcdef" | examples/static_iv.js:11:7:11:34 | fixedIV |
#select
| examples/static_iv.js:14:54:14:60 | fixedIV | examples/static_iv.js:11:17:11:34 | "0123456789abcdef" | examples/static_iv.js:14:54:14:60 | fixedIV | Insecure Initialization Vector (IV) used for cryptographic function. With a few exceptions, it is best to use a secure random source for IVs. |
